tests/yast2_gui/yast2_firewall_libyui.pm

```diff
@@ -0,0 +1,207 @@
+
+    #	enter page Custom Rules and check ui
+    assert_and_click("yast2_firewall_custom-rules");
+    # verify Custom Rules page is displayed
+    assert_screen("yast2_firewall_custom-rules-loaded");
+    send_key "alt-a";
+    assert_screen "yast2_firewall_add-new-allowing-rules";
+    send_key "alt-c";
+    wait_still_screen(2);
+
+    #	Next to finish and exit
+    send_key "alt-n";
+    assert_screen "yast2_firewall_summary", 30;
+    send_key "alt-f";
+}
+
+sub verify_service_stopped {
+
+    record_info('Start-Up', "Managing the firewalld service: Stop");
+    select_console 'x11', await_console => 0;
+    YaST::Module::open(module => 'firewall', ui => 'qt');
+    assert_screen 'yast2_firewall_start-up';
+    $testapi::distri->get_firewall()->get_main_page();
+    save_screenshot;
+    $testapi::distri->get_firewall()->get_start_up_page();
+    save_screenshot;
+    $testapi::distri->get_firewall()->stop_firewall();
+    save_screenshot;
+    $testapi::distri->get_firewall()->accept_change();
+    assert_screen 'generic-desktop';
+    select_console 'root-console';
+    assert_script_run("! (firewall-cmd --state) | grep 'not running'");
+}
+
+sub verify_service_started {
+
+    record_info('Start-Up', "Managing the firewalld service: Start");
+    select_console 'x11', await_console => 0;
+    YaST::Module::open(module => 'firewall', ui => 'qt');
+    assert_screen 'yast2_firewall_start-up';
+    $testapi::distri->get_firewall()->get_start_up_page();
+    save_screenshot;
+    $testapi::distri->get_firewall()->start_firewall();
+    save_screenshot;
+    $testapi::distri->get_firewall()->accept_change();
+    assert_screen 'generic-desktop';
+    select_console 'root-console';
+    assert_script_run "firewall-cmd --state | grep running";
+}
+
+sub verify_interface {
+    my (%args) = @_;
+    if ($testapi::distri->get_firewall()->verify_interface($args{device}, $args{zone}) != 1) {
+        die("Verify interface failed" . $args{device} . " " . $args{zone});
+    } else {
+        record_info('Verify interface ok', $args{device} . " " . $args{zone});
+    }
+}
+
+sub change_interface_zone {
+    my (%args) = @_;
+    $testapi::distri->get_firewall()->get_interfaces_page();
+    save_screenshot;
+    $testapi::distri->get_firewall()->set_interface_zone($args{device}, $args{zone});
+    save_screenshot;
+}
+
+sub verify_zone {
+    my (%args) = @_;
+
+    my $interfaces = $args{interfaces} //= 'no_interfaces';
+    my $default = $args{default} //= 'no_default';
+    my $menu_selected = $args{menu_selected} //= 0;
+    $testapi::distri->get_firewall()->get_zones_page();
+    save_screenshot;
+    if ($testapi::distri->get_firewall()->verify_zone($args{name}, $interfaces, $default) != 1) {
+        die("Verify zone failed! " . $args{name} . " " . $args{device} . " " . $interfaces . " " . $default);
+    } else {
+        record_info('verify zone ok! ', $args{name} . " " . $args{device} . " " . $interfaces . " " . $default);
```

Comment:
original test module abuses of using `record_info`, we should consider later when it makes sense because actually doesn't help much when there are too many, a well designed test doesn't need much use of that or even none.

Reply:
Yes, it like enable verbose log, but it help to narrow down issue once fail happen.